tools/purge_non_existing_images.sh

```diff
@@ -1,25 +1,71 @@
 #!/bin/sh
 
-DRYRUN=yes
+#
+# Usage: purge_non_existing_images [-p]
+#        -p  do the purge, otherwise only display non existing images
+#
 
-if [ "$1" = "-p" ]; then
-    DRYRUN=no
+if ! which sqlite3 > /dev/null; then
+    echo "error: please install sqlite3 binary".
+    exit 1
 fi
 
-DBFILE=$HOME/.config/darktable/library.db
-TMPFILE=$(mktemp -t tmp.XXXXXXXXXX)
-QUERY="select A.id,B.folder,A.filename from images as A join film_rolls as B on A.film_id = B.id"
+configdir="$HOME/.config/darktable"
+DBFILE="$configdir/library.db"
+dryrun=1
+library=""
 
-if ! which sqlite3 > /dev/null; then
-    echo error: please install sqlite3 binary.
-    exit 1
+# remember the command line to show it in the end when not purging
+commandline="$0 $*"
+
+# handle command line arguments
+while [ "$#" -ge 1 ] ; do
+  option="$1"
+  case ${option} in
+  -h|--help)
+    echo "Delete non existing images from darktable's database"
+    echo "Usage:   $0 [options]"
+    echo ""
+    echo "Options:"
+    echo "  -c|--configdir <path>    path to the darktable config directory"
+    echo "                           (default: '${configdir}')"
+    echo "  -l|--library <path>      path to the library.db"
+    echo "                           (default: '${DBFILE}')"
+    echo "  -p|--purge               actually delete the tags instead of just finding them"
+    exit 0
+    ;;
+  -l|--library)
+    library="$2"
+    shift
+    ;;
+  -c|--configdir)
+    configdir="$2"
+    shift
+    ;;
+  -p|--purge)
+    dryrun=0
+    ;;
+  *)
+    echo "warning: ignoring unknown option $option"
+    ;;
+  esac
+    shift
+done
+
+DBFILE="$configdir/library.db"
+
+if [ "$library" != "" ]; then
+    DBFILE="$library"
 fi
 
 if [ ! -f "$DBFILE" ]; then
-    echo error: library.db not found \($DBFILE\).
+    echo "library '${DBFILE}' doesn't exist"
```

Comment:
For consistency I think all error message should start with "error: "

Reply:
Yes! Will do later today :)